### DISCUSSION TITLE
very odd duplicate being sent by gmail when accepting a calendar invite

### DISCUSSION THREAD

**Discussion by redstang33** [kewisch/gdata-provider#314] [Support] [answered]
Hi,

I have seen this happen twice now and it's embarassing to me. Here's what happens:
- I get a calendar invite in my (non-google) email.
- I accept the invite in Thunderbird to my google calendar.
- I get a pop-up "Party crashing? You are not on the guest list yet. Respond as:" and I choose the email address the invite was sent to. Note that at this point my private email is indeed on the guest list, however not the gmail account that I accept the invite into.
- The sender receives the confirmation e-mail that I accepted the entry

HOWEVER: At this point EVERYBODY that was invited to the original invite receives a NEW invite from my google-account I used to accept the invite into duplicating the original invite.

This only started to happen with Thunderbird 91.1.2 that got installed with a recent Ubuntu upgrade I'm using. Hence it might very well be related to Thunderbird and not the gdata-provider plug-in.

**Comment by nverwer** [comment]
The same happens to me on Windows, after a recent Thunderbird update.

**Comment by kewisch** [comment]
I'd be interested in a debug log that shows what happens on every step of the way. See https://github.com/kewisch/gdata-provider/wiki/FAQ#enabling-debugging for details.

**Comment by redstang33** [comment]
Steps to reproduce:
- Sent an invite from an external address to your email that you have in Thunderbird.
- Accept the meeting into a google calendar (different email address):
  - Select Calendar dialog
  - Party crashing dialog (You are not on the guest list yet. Respond as: use primary email).
  
Response is sent. At this point the calendar entry can be seen in Thunderbird Calendar as being from the correct sender. The original sender receives the accepted confirmation - sent from Thunderbird.
At this point the invite is NOT visible on google calendar web-interface though!

Timestamps up to 10:18:46.

Then for a while nothing happens!

I restarted Thunderbird now. (10:33)

Now the google calendar entry is visible on the web-interface. Organizer is the calendar the request got accepted into and invitation emails were sent from google.

I don't always have to restart Thunderbird to trigger this but there is always a delay in accepting the invite until the new entry is sent from google.

I hope the logs are helpful. I had to edit them to not reveal too much personal details. I think I caught the creation of the original entry and then the creation of the duplicate with the sender being the owner of the calendar the event was imported into.

Sender: sender@another.address
Recipient email: primary@email.address
Recipient Calendar the entry is accepted into: primary_calendar@gmail.com

... hope this all helps!






10:17:39.472 Calendar: iTIP method: REQUEST calItipUtils.jsm:222
10:17:39.532 Calendar: iTIP on REQUEST: no existing items. calItipUtils.jsm:1629
10:17:39.533 Calendar: iTIP operations: 1 calItipUtils.jsm:1716
10:17:39.533 Calendar: iTIP options on: REQUEST calItipUtils.jsm:400
10:18:43.712 Calendar: Ignoring attendee duplicate for item redacted (Lightning test): mailto:primary@email.address calItemBase.js:620
10:18:43.724 Calendar: [calGoogleCalendar] Adding invitation event Lightning test gdataCalendar.jsm:406
10:18:43.725 Calendar: [calGoogleRequest] Requesting POST https://www.googleapis.com/calendar/v3/calendars/primary_calendar%40gmail.com/events/import?sendNotifications=true gdataRequest.jsm:212
10:18:43.725 Calendar: [calGoogleCalendar] Setting Upload Data (application/json; charset=UTF-8):
{"start":{"dateTime":"2021-11-22T13:30:00","timeZone":"Europe/Berlin"},"end":{"dateTime":"2021-11-22T14:00:00","timeZone":"Europe/Berlin"},"iCalUID":"redacted","updated":"2021-11-22T09:18:43.000Z","summary":"Lightning test","description":"\n","transparency":"opaque","visibility":"public","sequence":"0","status":"confirmed","organizer":{"email":"sender@another.address","displayName":"Lastname, Firstname"},"attendees":[{"email":"primary@email.address","displayName":"Firstname Lastname","responseStatus":"accepted"},{"email":"sender@another.address","displayName":"Lastname, Firstname"}],"reminders":{"overrides":[],"useDefault":false}} gdataRequest.jsm:279
10:18:43.831 POSThttps://www.googleapis.com/calendar/v3/calendars/primary_calendar@gmail.com/events/import?sendNotifications=true
[HTTP/3 400 Bad Request 153ms]

10:18:44.044 Calendar: [calGoogleRequest] Clock skew is 1 seconds gdataRequest.jsm:382
10:18:44.044 Calendar: [calGoogleCalendar] Request POST https://www.googleapis.com/calendar/v3/calendars/primary_calendar%40gmail.com/events/import?sendNotifications=true responded with HTTP 400 gdataRequest.jsm:389
10:18:44.045 Calendar: [calGoogleCalendar] A request Error Occurred. Status Code: 400 Bad Request Body: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "participantIsNeitherOrganizerNorAttendee",
    "message": "The owner of the calendar must either be the organizer or an attendee of an event that is imported."
   }
  ],
  "code": 400,
  "message": "The owner of the calendar must either be the organizer or an attendee of an event that is imported."
 }
}
gdataRequest.jsm:540
10:18:44.046 Calendar: [calGoogleCalendar] Adding Item Lightning test failed:2147746065: A request Error Occurred. Status Code: 400 Bad Request Body: {
 "error": {
  "errors": [
   {
    "domain": "calendar",
    "reason": "participantIsNeitherOrganizerNorAttendee",
    "message": "The owner of the calendar must either be the organizer or an attendee of an event that is imported."
   }
  ],
  "code": 400,
  "message": "The owner of the calendar must either be the organizer or an attendee of an event that is imported."
 }
}
gdataCalendar.jsm:457
10:18:44.046 Calendar: [calCachedCalendar] Calendar GCalendar is unavailable, adding item offline calCachedCalendar.js:762
10:18:44.144 Calendar: iTIP on REQUEST: found 1 items. calItipUtils.jsm:1377
10:18:44.147 Calendar: iTIP operations: 0 calItipUtils.jsm:1716
10:18:44.157 Calendar: sendItems: Preparing to send an invitation email... CalItipEmailTransport.jsm:402
10:18:44.158 Calendar: sendXpcomMail: Found AUTO autoResponse type. CalItipEmailTransport.jsm:199
10:18:44.161 Calendar: mail text:
MIME-version: 1.0

From: Firstname Lastname <primary@email.address>

Message-ID: <549f60a0-b070-f540-1af8-d83e88d035ec@email.address>

To: "Lastname, Firstname" <sender@another.address>

Date: Mon, 22 Nov 2021 10:18:44 +0100

Subject: Accepted: Lightning test

Content-type: multipart/mixed; boundary="Boundary_(ID_qyG4ZdjoAsiZ+Jo19dCbWQ)"





--Boundary_(ID_qyG4ZdjoAsiZ+Jo19dCbWQ)

Content-type: multipart/alternative;

 boundary="Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)"





--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)

Content-type: text/plain; charset=UTF-8

Content-transfer-encoding: 8BIT



Firstname Lastname <primary@email.address> has accepted your event invitation.





--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)

Content-type: text/calendar; method=REPLY; charset=UTF-8

Content-transfer-encoding: 8BIT



BEGIN:VCALENDAR

PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN

VERSION:2.0

METHOD:REPLY

BEGIN:VTIMEZONE

TZID:Europe/Berlin

BEGIN:DAYLIGHT

TZOFFSETFROM:+0100

TZOFFSETTO:+0200

TZNAME:CEST

DTSTART:19700329T020000

RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3

END:DAYLIGHT

BEGIN:STANDARD

TZOFFSETFROM:+0200

TZOFFSETTO:+0100

TZNAME:CET

DTSTART:19701025T030000

RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10

END:STANDARD

END:VTIMEZONE

BEGIN:VEVENT

LAST-MODIFIED:20211122T091844Z

DTSTAMP:20211122T091844Z

UID:<redacted>

SUMMARY:Lightning test

PRIORITY:5

STATUS:CONFIRMED

ORGANIZER;CN="Lastname, Firstname":mailto:sender@another.address

ATTENDEE;CN=Firstname Lastname;PARTSTAT=ACCEPTED;CUTYPE=INDIVIDUAL:mailto:primary@email.address

DTSTART;TZID=Europe/Berlin:20211122T133000

DTEND;TZID=Europe/Berlin:20211122T140000

DESCRIPTION;LANGUAGE=en-US:\n

CLASS:PUBLIC

TRANSP:OPAQUE

SEQUENCE:0

X-MICROSOFT-CDO-APPT-SEQUENCE:0

X-MICROSOFT-CDO-OWNERAPPTID:-<redacted>

X-MICROSOFT-CDO-BUSYSTATUS:TENTATIVE

X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY

X-MICROSOFT-CDO-ALLDAYEVENT:FALSE

X-MICROSOFT-CDO-IMPORTANCE:1

X-MICROSOFT-CDO-INSTTYPE:0

X-MICROSOFT-DONOTFORWARDMEETING:FALSE

X-MICROSOFT-DISALLOW-COUNTER:FALSE

X-MICROSOFT-LOCATIONS:[]

END:VEVENT

END:VCALENDAR





--Boundary_(ID_ryU4ZdJoASiZ+Jo21dCbwA)--



--Boundary_(ID_qyG4ZdjoAsiZ+Jo19dCbWQ)

Content-type: application/ics; name=invite.ics

Content-transfer-encoding: 8BIT

Content-disposition: attachment; filename=invite.ics



BEGIN:VCALENDAR

PRODID:-//Mozilla.org/NONSGML Mozilla Calendar V1.1//EN

VERSION:2.0

METHOD:REPLY

BEGIN:VTIMEZONE

TZID:Europe/Berlin

BEGIN:DAYLIGHT

TZOFFSETFROM:+0100

TZOFFSETTO:+0200

TZNAME:CEST

DTSTART:19700329T020000

RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=3

END:DAYLIGHT

BEGIN:STANDARD

TZOFFSETFROM:+0200

TZOFFSETTO:+0100

TZNAME:CET

DTSTART:19701025T030000

RRULE:FREQ=YEARLY;BYDAY=-1SU;BYMONTH=10

END:STANDARD

END:VTIMEZONE

BEGIN:VEVENT

LAST-MODIFIED:20211122T091844Z

DTSTAMP:20211122T091844Z

UID:<redacted>

SUMMARY:Lightning test

PRIORITY:5

STATUS:CONFIRMED

ORGANIZER;CN="Lastname, Firstname":mailto:sender@another.address

ATTENDEE;CN=Firstname Lastname;PARTSTAT=ACCEPTED;CUTYPE=INDIVIDUAL:mailto:primary@email.address

DTSTART;TZID=Europe/Berlin:20211122T133000

DTEND;TZID=Europe/Berlin:20211122T140000

DESCRIPTION;LANGUAGE=en-US:\n

CLASS:PUBLIC

TRANSP:OPAQUE

SEQUENCE:0

X-MICROSOFT-CDO-APPT-SEQUENCE:0

X-MICROSOFT-CDO-OWNERAPPTID:-<redacted>

X-MICROSOFT-CDO-BUSYSTATUS:TENTATIVE

X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY

X-MICROSOFT-CDO-ALLDAYEVENT:FALSE

X-MICROSOFT-CDO-IMPORTANCE:1

X-MICROSOFT-CDO-INSTTYPE:0

X-MICROSOFT-DONOTFORWARDMEETING:FALSE

X-MICROSOFT-DISALLOW-COUNTER:FALSE

X-MICROSOFT-LOCATIONS:[]

END:VEVENT

END:VCALENDAR





--Boundary_(ID_qyG4ZdjoAsiZ+Jo19dCbWQ)--

CalItipEmailTransport.jsm:339
10:18:44.162 Calendar: _createTempImipFile path: /tmp/itipTemp CalItipEmailTransport.jsm:361
10:18:46.439 Calendar: No unique email address for lookup! 2 calItipUtils.jsm:1038
10:27:00.520 Calendar: iTIP method: REQUEST calItipUtils.jsm:222
10:27:00.674 Calendar: iTIP on REQUEST: found 1 items. calItipUtils.jsm:1377
10:27:00.678 Calendar: iTIP operations: 0 calItipUtils.jsm:1716
10:27:04.444 Calendar: iTIP method: REQUEST calItipUtils.jsm:222
10:27:04.476 Calendar: iTIP on REQUEST: found 1 items. calItipUtils.jsm:1377
10:27:04.479 Calendar: iTIP operations: 0 calItipUtils.jsm:1716



10:33:07.827 Calendar: [CalSleepMonitor] Starting sleep monitor. CalSleepMonitor.jsm:57
10:33:08.165 Unexpected event profile-after-change URLQueryStrippingListService.jsm:224
10:33:08.173 Calendar: [CalTimezoneService] Loading resource:///res/zones.json CalTimezoneService.jsm:87
10:33:09.062 Calendar: [CalTimezoneService] Timezones version 2.2021c loaded CalTimezoneService.jsm:133
10:33:10.431 Unknown Collection "thunderbird/query-stripping" RemoteSettingsClient.jsm:160
10:33:10.447 Calendar: [calGoogleSessionManager] Creating session primary_calendar@gmail.com gdataSession.jsm:88
10:33:10.573 Calendar: [calGoogleSessionManager] Reusing session primary_calendar@gmail.com 20 gdataSession.jsm:86
10:33:10.967 Successfully loaded OpenPGP library librnp.so version 0.15.2+git20210806.dd923a4e.MZLA from /usr/lib/thunderbird/librnp.so RNPLib.jsm:92:15
10:33:11.302 Found 95 public keys and 6 secret keys (6 protected, 0 unprotected) RNPLib.jsm:288:15
10:33:11.304 Successfully loaded optional OpenPGP library libgpgme.so.11 from system's standard library locations GPGMELib.jsm:69:13
10:33:11.305 gpgme version: 1.14.0-unknown GPGMELib.jsm:241:15
10:33:11.307 Assert failed: Calendar manager initialised calendars before loadCalendarComponent ran on the first 3pane window. This should not happen.
 calUtils.jsm:121
10:33:11.354 Trying to load /usr/lib/thunderbird/libotr.so OTRLib.jsm:64:11
10:33:11.355 Trying to load libotr.so from system's standard library locations OTRLib.jsm:64:11
10:33:11.355 Trying to load libotr.so.5 from system's standard library locations OTRLib.jsm:64:11
10:33:11.359 Successfully loaded OTR library libotr.so.5 from system's standard library locations OTRLib.jsm:72:13
10:33:12.024 tb.account.size_on_disk - Attempted to set the scalar to an incompatible value. 2
10:33:12.025 tb.account.size_on_disk - Trying to set an unsigned scalar to a negative number. 2
10:33:12.059 Calendar: Start loading of locale dependent preference default values... calendar-chrome-startup.js:389
10:33:12.063 Calendar: Loading of locale sensitive preference default values completed. calendar-chrome-startup.js:409

10:33:13.400 services.settings: thunderbird/hijack-blocklists has signature disabled RemoteSettingsClient.jsm:1027

10:33:13.733 Calendar: [calCachedCalendar] Performing playback operation add on 1 items to GCalendar calCachedCalendar.js:648
10:33:13.734 Calendar: [calGoogleCalendar] Adding regular event Lightning test gdataCalendar.jsm:409
10:33:13.734 Calendar: [calGoogleSession] Token expired 1637573594 seconds ago, resetting gdataSession.jsm:388
10:33:13.734 Calendar: [calGoogleCalendar] Logging in session primary_calendar@gmail.com gdataSession.jsm:266
10:33:13.735 Calendar: [calGoogleCalendar] No access token for primary_calendar@gmail.com, refreshing token gdataSession.jsm:347
10:33:13.735 Calendar: [calGoogleSession] Adding item https://www.googleapis.com/calendar/v3/calendars/primary_calendar%40gmail.com/events to queue gdataSession.jsm:407
10:33:13.879 Calendar: [calGoogleCalendar] Successfully acquired a new OAuth token for primary_calendar@gmail.com gdataSession.jsm:270
10:33:13.879 Calendar: [calGoogleRequest] Requesting POST https://www.googleapis.com/calendar/v3/calendars/primary_calendar%40gmail.com/events?sendNotifications=true gdataRequest.jsm:212
10:33:13.879 Calendar: [calGoogleCalendar] Setting Upload Data (application/json; charset=UTF-8):
{"start":{"dateTime":"2021-11-22T13:30:00","timeZone":"Europe/Berlin"},"end":{"dateTime":"2021-11-22T14:00:00","timeZone":"Europe/Berlin"},"summary":"Lightning test","description":"\n","transparency":"opaque","visibility":"public","sequence":"0","status":"confirmed","organizer":{"email":"sender@another.address","displayName":"Lastname, Firstname"},"attendees":[{"email":"primary@email.address","displayName":"Firstname Lastname","responseStatus":"accepted"},{"email":"sender@another.address","displayName":"Lastname, Firstname"}],"reminders":{"overrides":[],"useDefault":false}} gdataRequest.jsm:279
10:33:14.098 Calendar: [calCachedCalendar] Doing changelog based sync for calendar googleapi://primary_calendar@gmail.com/?calendar=redacted%40import.calendar.google.com calCachedCalendar.js:340
10:33:14.104 Calendar: [calGoogleRequest] Requesting GET https://www.googleapis.com/calendar/v3/users/me/calendarList…redacted%40import.calendar.google.com gdataRequest.jsm:212
10:33:14.106 Calendar: [calGoogleRequest] Requesting GET https://www.googleapis.com/calendar/v3/calendars/redacted…eleted=true&syncToken=redacted gdataRequest.jsm:212
10:33:14.234 Calendar: [calGoogleCalendar] Request POST https://www.googleapis.com/calendar/v3/calendars/primary_calendar%40gmail.com/events?sendNotifications=true responded with HTTP 200 gdataRequest.jsm:389
10:33:14.234 Calendar: [calGoogleCalendar] Parsing entry:
{
 "kind": "calendar#event",
 "etag": "\"redacted\"",
 "id": "redacted",
 "status": "confirmed",
 "htmlLink": "https://www.google.com/calendar/event?eid=redacted",
 "created": "2021-11-22T09:33:13.000Z",
 "updated": "2021-11-22T09:33:14.050Z",
 "summary": "Lightning test",
 "description": "\n",
 "creator": {
  "email": "primary_calendar@gmail.com",
  "self": true
 },
 "organizer": {
  "email": "primary_calendar@gmail.com",
  "self": true
 },
 "start": {
  "dateTime": "2021-11-22T13:30:00+01:00",
  "timeZone": "Europe/Berlin"
 },
 "end": {
  "dateTime": "2021-11-22T14:00:00+01:00",
  "timeZone": "Europe/Berlin"
 },
 "visibility": "public",
 "iCalUID": "redacted@google.com",
 "sequence": 0,
 "attendees": [
  {
   "email": "sender@another.address",
   "displayName": "Lastname, Firstname",
   "responseStatus": "needsAction"
  },
  {
   "email": "primary@email.address",
   "displayName": "Firstname Lastname",
   "responseStatus": "accepted"
  }
 ],
 "reminders": {
  "useDefault": false
 },
 "eventType": "default"
}
gdataLogging.jsm:16
10:33:14.572 Calendar: [calGoogleCalendar] Request GET https://www.googleapis.com/calendar/v3/users/me/calendarList…redacted%40import.calendar.google.com responded with HTTP 200 gdataRequest.jsm:389
10:33:14.581 Calendar: [calGoogleCalendar] Request GET https://www.googleapis.com/calendar/v3/calendars/redacted…eleted=true&syncToken=CJjn3LDPqvQCEJjn3LDPqvQCGAEgkMPrugE%3D responded with HTTP 200 gdataRequest.jsm:389
10:33:14.582 Calendar: [calGoogleCalendar] Timezone for FC Wädenswil - Junioren C 2. Stärkeklasse c is Europe/Amsterdam gdataUtils.jsm:1098
10:33:14.582 Calendar: [calCachedCalendar] replayChangesOn finished. calCachedCalendar.js:357
10:33:14.588 Calendar: [calCachedCalendar] sync queue empty. calCachedCalendar.js:329
10:33:14.606 Calendar: [calGoogleCalendar] Adding Lightning test succeeded gdataCalendar.jsm:451

**Comment by kewisch** [comment]
Hmm, the telling line may be `Calendar GCalendar is unavailable, adding item offline`. The event is saved in the cache, and then later replayed. I'm not sure why the replayed item would have an organizer that doesn't trigger the same 400 error, this would be interesting.

One way to reproduce this quicker is to switch Thunderbird into offline mode, and then go online again. The cached items should then be replayed, so you can capture what is sent to the server at that point.

A potential fix may be to change code in the provider that if this type of 400 error is sent, it is registered as a hard failure, instead of the server being offline. I'm also interested in why Thunderbird is deciding to send an email manually though, not quite sure how to debug this yet.

**Comment by kewisch** [comment, marked as answer]
I created #347 to track this issue, let's continue there.
